slash/parallel/worker_configuration.py

```diff
@@ -1,3 +1,137 @@
+from ..conf import config
+from .. import hooks
+import os
+import logbook
+from ..utils.tmux_utils import create_new_window, create_new_pane
+from slash import ctx
+import subprocess
+import time
+import signal
+import errno
+import sys
+
+_logger = logbook.Logger(__name__)
+
+def is_process_running(pid):
+    try:
+        os.kill(pid, 0)
+    except OSError as err:
+        if err.errno == errno.ESRCH:
+            return False
+        else:
+            raise
+    return True
+
 class WorkerConfiguration(object):
-    def __init__(self, argv):
-        self.argv = argv
+    def __init__(self, args, worker_id):
+        self._worker_id = worker_id
+        self.argv = [sys.executable, '-m', 'slash.frontend.main', 'run', '--parallel-parent-session-id', ctx.session.id] + \
+                     args + ["--parallel-worker-id", str(worker_id)]
+        self._excluded_tests = set()
+        self._forced_tests = set()
+        self._pid = None
+
+    def get_forced_tests(self):
+        return self._forced_tests
+
+    def get_excluded_tests(self):
+        return self._excluded_tests
+
+    def get_worker_id(self):
+        return self._worker_id
+
+    def _start(self):
+        raise NotImplementedError()
+
+    def start(self):
+        hooks.before_worker_start(worker_config=self) # pylint: disable=no-member
+        _logger.notice("Starting worker number {}", self._worker_id)
+        self.argv += ['--parallel-port', str(ctx.session.parallel_manager.server.port),\
+                      '--keepalive-port', str(ctx.session.parallel_manager.keepalive_server.port), \
+                      '--workers-error-dir', ctx.session.parallel_manager.workers_error_dircetory]
+        self._start()
+
+    def kill(self):
+        raise NotImplementedError()
+
+    def handle_timeout(self):
+        raise NotImplementedError()
+
+    def exclude_test(self, test_index):
```

Comment:
Add docstring to this, explaining semantics

Reply:
I would also make this method receive either a test object or a number (to make it easy for plugins/hook implementations)